### PR TITLE
Create source tarball

### PR DESCRIPTION
**Pull request by hydai**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,12 @@ if(GIT_VERSION_NOT_FOUND AND NOT GIT_VERSION_NOT_FOUND EQUAL 0)
   set(CPACK_PACKAGE_VERSION "0.0.0-unreleased")
 endif()
 
+# Overwrite version information once there is a VERSION file
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/VERSION")
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/VERSION" LOCAL_VERSION)
+  set(CPACK_PACKAGE_VERSION LOCAL_VERSION)
```
Comment (ibmibmibm) [inline CMakeLists.txt:23]: `unset(LOCAL_VERSION)`

**Review by ibmibmibm — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -19,6 +19,43 @@ on:
...
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Get version
+      id: prep
+      run: |
+        echo ::set-output name=version::$(git describe --match "[0-9].[0-9]*" --tag)
+    - name: Propagate version information for tarball
+      run: |
+        echo $VERSION > VERSION
```
[the remark this answers is not in the document]
Comment (hydai) [reply .github/workflows/build.yml]: Fixed

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -19,6 +19,43 @@ on:
...
+      env:
+        VERSION: ${{ steps.prep.outputs.version }}
+    - name: Verify the version information
+      run: |
+        cat VERSION
```
[the remark this answers is not in the document]
Comment (hydai) [reply .github/workflows/build.yml]: Fixed

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -19,6 +19,43 @@ on:
...
+    - name: Create source tarball
+      run: |
+        TEMPDIR=$(mktemp -d)
+        SRCDIR="$TEMPDIR/wasmedge/"
+        mkdir -p "$SRCDIR"
+        git checkout-index -a --prefix="$SRCDIR"
+        cp VERSION $SRCDIR
```
[the remark this answers is not in the document]
Comment (hydai) [reply .github/workflows/build.yml]: Fixed

File: CMakeLists.txt (hunk 1 above)
Comment (hydai) [reply CMakeLists.txt:23]: Fixed

**Review by ibmibmibm — DISMISSED** [review]

**Review by ibmibmibm — APPROVED** [review]

**Pull request merged**